### PR TITLE
chore: add snapshot testing using jest

### PR DESCRIPTION
**Pull request by Flui**
## Description
Move from mocha/chai/sinon to jest.

## Motivation and Context
To get snapshot tests and to have one framwork for unit-test, snapshot-tests including function mocking.

## How Has This Been Tested?
Ran all tests.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,9 +61,9 @@
     "build-docs": "./build_helpers/buildAPIDocs.sh",
     "publish-site": "./build_helpers/publishStaticSite.sh",
     "publish-package": "./build_helpers/publishPackage.sh",
-    "test": "mocha-webpack --webpack-config webpack.config-test.js \"src/**/*-test.js\" --require build_helpers/test-globals.js",
-    "test:watch": "mocha-webpack --webpack-config webpack.config-test.js --watch \"src/**/*-test.js\" --require build_helpers/test-globals.js",
-    "test:server": "webpack-dev-server --config webpack.config-test.js --hot --inline"
```
Comment (Flui) [inline package.json]: I do not know how important this feature is to run the tests in the browser. But as there is no such thing for jest I deleted this task. Actually I could also remove webpack.config-test.js

**Comment by Flui** [comment]
As jest v22 does not support node 4 and 5 I decided to choose jest v21

**Comment by wcjordan** [comment]
Great thanks for adding in Jest snapshot testing!  Since we're not interested in switching the mocha tests to use jest at this time, could you switch those back and modify the test target to run both test runners?

**Comment by wcjordan** [comment]
Any luck getting the tests fixed?  I was able to get jest working by adding "rootDir": "src" to the jest config in package.json.  The other tests appear to need some fixes as well though.

**Comment by Flui** [comment]
For the last test I need some help. It seems as this is caused by the shallow rendering, which is an indicator that this component might not be encapsulated enough

**Comment by wcjordan** [comment]
It looks like you're missing a some props for the component.  In particular the error I see is from missing the tableSize prop.  Unfortunately the PropTypes there are still being used to generate the documentation rather than for runtime assertions.

I recommend checking out [FixedDataTableContainer](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/FixedDataTableContainer.js) and the [default reducer state](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/reducers/index.js#L24) for an idea of the props you need.

**Comment by Flui** [comment]
But why aren't you using the proptypes? I think it is easier to debug the code if you use proptypes or a static type checker like flow or typescript

**Comment by wcjordan** [comment]
We are, they just haven't been updated on the beta branch with the new changes yet.  Updating them will also require changes to how react-docgen is being used & the docs so we continue to generate user facing props in the documentation (now passed into FDTContainer).

**Comment by wcjordan** [comment]
Closing this PR as stale.  @pradeepnschrodinger is there anything here you think we should take and see if we can get it working with the current master?

**Pull request closed without merging**